### DISCUSSION TITLE
Sequel.| and Sequel.& have an unexpected behavior

### DISCUSSION THREAD

**Discussion by ngaumont** [jeremyevans/sequel#1905] [General]
Hi,

I'm using the Sequel.| operator and found an expected behavior.
Using an [example](http://sequel.jeremyevans.net/rdoc/files/doc/dataset_filtering_rdoc.html#label-Filtering+using+expressions) from the doc, this returns  the expected SQL:

```ruby
items.where{Sequel.|({category: 'ruby'}, (price + 100 < 200))}.sql
# "SELECT * FROM items WHERE ((category = 'ruby') OR ((price + 100) < 200))"
```
However, the following does not crash and returns an unexpected SQL:

```ruby
items.where{Sequel.&({category: not_defined}, (price + 100 < 200))}.sql
# "SELECT * FROM items WHERE ((category = 'not_defined') AND ((price + 100) < 200))"
```
I don't understand how this codes runs with `not_defined` being not defined.  
I did read the [implementation](https://github.com/jeremyevans/sequel/blob/master/lib/sequel/sql.rb#L848), but I have no clue.

Is it an intended feature ? What am I missing ?

Thanks

**Comment by jeremyevans** [comment]
Note sure how `not_defined` is showing up as a single quoted string.  You should be getting an identifier.  That's what I get when I test the code. The identifier doesn't come from Sequel.&, it comes from the that that you are using where with a virtual row block.  Because `not_defined` is not a local variable in scope, it is treated as a method, and handled by the virtual row block's method_missing, the same way that `price` is treated.

**Reply by ngaumont** [reply]
You are completely right. This problem was linked to the block.
Thank you for the explanation.
